src/all/deviantart/src/eu/kanade/tachiyomi/extension/all/deviantart/DeviantArt.kt

```diff
@@ -0,0 +1,187 @@
+package eu.kanade.tachiyomi.extension.all.deviantart
+
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.HttpSource
+import eu.kanade.tachiyomi.util.asJsoup
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.jsonObject
+import kotlinx.serialization.json.jsonPrimitive
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Request
+import okhttp3.Response
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Document
+import org.jsoup.parser.Parser
+import java.text.ParseException
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+class DeviantArt : HttpSource() {
+    override val name = "DeviantArt"
+    override val baseUrl = "https://deviantart.com"
+    override val lang = "all"
+    override val supportsLatest = false
+
+    private val backendBaseUrl = "https://backend.deviantart.com"
+    private fun backendBuilder() = backendBaseUrl.toHttpUrl().newBuilder()
+
+    private val dateFormat by lazy {
+        SimpleDateFormat("EEE, dd MMM yyyy HH:mm:ss zzz", Locale.ENGLISH)
+    }
+
+    private fun parseDate(dateStr: String): Long {
+        return try {
+            dateFormat.parse(dateStr)!!.time
+        } catch (_: ParseException) {
+            0L
+        }
+    }
+
+    override fun popularMangaRequest(page: Int): Request {
+        throw UnsupportedOperationException(SEARCH_ERROR_MSG)
+    }
+
+    override fun popularMangaParse(response: Response): MangasPage {
+        throw UnsupportedOperationException(SEARCH_ERROR_MSG)
+    }
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val url = backendBuilder()
+            .addPathSegment("rss.xml")
+            .addQueryParameter("q", query)
+            .build()
+
+        return GET(url, headers)
+    }
+
+    override fun searchMangaParse(response: Response): MangasPage {
+        val document = response.asJsoupXml()
+        requireNotNull(document.selectFirst("item")) { SEARCH_ERROR_MSG }
+
+        val (username, folderId) = response.request.url.queryParameter("q")!!.split(" ")
+            .find { it.startsWith("gallery:") }
+            ?.substringAfter(":")
+            ?.split("/")
+            ?: throw IllegalArgumentException(SEARCH_ERROR_MSG)
+
+        val newRequest = GET("$baseUrl/$username/gallery/$folderId", headers)
```

Comment:
do this check in `searchMangaRequest` and return this request there, since the rss response isn't really being used other that to throw an exception

Reply:
Ah I didn't notice it was no longer necessary, thanks